#! /bin/sh /usr/share/dpatch/dpatch-run
## makefile_adjustments.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: Adjust the installation paths to Debian-compatible values, and drag
## DP: all install incovations to the install target because $(DESTDIR)
## DP: (i.e. debian/ebtables) isn't available before.
## DP:
## DP: Also fix an FTBFS bug on HPPA by calling $(CC) instead of $(LD).

@DPATCH@
Index: ebtables-2.0.9.2/Makefile
===================================================================
--- ebtables-2.0.9.2.orig/Makefile	2010-02-03 22:17:45.000000000 +0100
+++ ebtables-2.0.9.2/Makefile	2010-05-07 16:46:47.000000000 +0200
@@ -8,11 +8,11 @@
 
 # default paths
 LIBDIR:=/usr/lib
-MANDIR:=/usr/local/man
-BINDIR:=/usr/local/sbin
+MANDIR:=/usr/share/man
+BINDIR:=/sbin
 ETCDIR:=/etc
-INITDIR:=/etc/rc.d/init.d
-SYSCONFIGDIR:=/etc/sysconfig
+INITDIR:=/etc/init.d
+SYSCONFIGDIR:=/etc/default
 DESTDIR:=
 
 CFLAGS:=-Wall -Wunused
@@ -154,18 +154,11 @@
 .PHONY: scripts
 scripts: ebtables-save ebtables.sysv ebtables-config
 	cat ebtables-save | sed 's/__EXEC_PATH__/$(tmp1)/g' > ebtables-save_
-	install -m 0755 -o root -g root ebtables-save_ $(DESTDIR)$(BINDIR)/ebtables-save
 	cat ebtables.sysv | sed 's/__EXEC_PATH__/$(tmp1)/g' | sed 's/__SYSCONFIG__/$(tmp2)/g' > ebtables.sysv_
-	install -m 0755 -o root -g root ebtables.sysv_ $(DESTDIR)$(INITDIR)/ebtables
 	cat ebtables-config | sed 's/__SYSCONFIG__/$(tmp2)/g' > ebtables-config_
-	install -m 0600 -o root -g root ebtables-config_ $(DESTDIR)$(SYSCONFIGDIR)/ebtables-config
-	rm -f ebtables-save_ ebtables.sysv_ ebtables-config_
 
 $(MANDIR)/man8/ebtables.8: ebtables.8
-	mkdir -p $(DESTDIR)$(@D)
 	sed 's/$$(VERSION)/$(PROGVERSION)/' ebtables.8 | sed 's/$$(DATE)/$(PROGDATE)/' > ebtables.8_
-	install -m 0644 -o root -g root ebtables.8_ $(DESTDIR)$@
-	rm -f ebtables.8_
 
 $(ETHERTYPESFILE): ethertypes
 	mkdir -p $(DESTDIR)$(@D)
@@ -179,9 +172,13 @@
 
 .PHONY: install
 install: $(MANDIR)/man8/ebtables.8 $(ETHERTYPESFILE) exec scripts
-	mkdir -p $(DESTDIR)$(LIBDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR) $(DESTDIR)$(INITDIR) $(DESTDIR)$(SYSCONFIGDIR) $(DESTDIR)$(MANDIR)/man8
 	install -m 0755 extensions/*.so $(DESTDIR)$(LIBDIR)
 	install -m 0755 *.so $(DESTDIR)$(LIBDIR)
+	install -m 0755 -o root -g root ebtables-save_ $(DESTDIR)$(BINDIR)/ebtables-save
+	install -m 0755 -o root -g root ebtables.sysv_ $(DESTDIR)$(INITDIR)/ebtables
+	install -m 0600 -o root -g root ebtables-config_ $(DESTDIR)$(SYSCONFIGDIR)/ebtables-config
+	install -m 0644 -o root -g root ebtables.8_ $(DESTDIR)$(MANDIR)/man8/ebtables.8
 
 .PHONY: clean
 clean:
